[allure-adaptor] Add ability to customize tab with defect categories (#5506)
vividus-framework/vividus: docs/modules/configuration/pages/reporting.adoc

report.brand.title=MY BRAND
----

== Configure defect categories

It is possible to customize the names of defect categories in the report, as well as the statuses of tests that will be categorized under these names. +

> Use the next properties:

Reviewer: ```suggestion
Use the following properties:
```

[source,properties]
----

> report.result.categories.<category-identifier>.name=<name>
> report.result.categories.<category-identifier>.statuses=<statuses>

Reviewer: ```suggestion
report.tabs.categories.<category-identifier>.name=<name>
report.tabs.categories.<category-identifier>.statuses=<statuses>
```

----

* `category-identifier` - The unique identifier of category.

> * `name` - The desired category name.

Reviewer: ```suggestion
* `name` - The desired category name to be displayed in the report.
```

* `statuses` - The comma-separated list of test statuses included in this category. Possible values:
https://allurereport.org/docs/test-statuses/#failed[failed],
https://allurereport.org/docs/test-statuses/#broken[broken],
unknown (in context of VIVIDUS - test cases with xref:tests-configuration.adoc#_known_issues[Known issues]),

> https://allurereport.org/docs/test-statuses/#skipped[skipped].

Reviewer: is it possible to map passed tests as well?

Author: > is it possible to map passed tests as well?

Yep, but I didn't include it in the doc because not quite sure that it is correct to include Passed tests in this context (defect categories)

Reviewer: just use `categories` instead of `defect categories`

[NOTE]
====

> User defect categories precedence over default categories. So if you create your own configuration for failed and broken tests,

Reviewer: ```suggestion
User defect categories take precedence over default categories. So if you create your own configuration for failed and broken tests,
```

but ignore test cases from `Known issues` category (`unknown` status), these defects will not be displayed in the  `Categories` tab.

> This rule does not apply for `Test defects`(`broken` status) and `Product defects`(`failed` status) categories as they are integral part of the report.

Reviewer: ```suggestion
This rule does not apply to `Test defects`(`broken` status) and `Product defects`(`failed` status) categories: if user configuration is not defined, the default categories will be shown.
```
